connectors/sources/salesforce.py

```diff
@@ -147,13 +232,101 @@ def with_default_metafields(self):
     def with_fields(self, fields):
         self.fields.extend(fields)
 
+    def with_where(self, where_string):
+        self.where = f"WHERE {where_string}"
+
+    def with_order_by(self, order_by_string):
+        self.order_by = f"ORDER BY {order_by_string}"
+
+    def with_limit(self, limit):
+        self.limit = f"LIMIT {limit}"
+
+    def with_join(self, join):
+        self.fields.append(f"(\n{join})\n")
+
     def build(self):
         select_columns = ",\n".join(set(self.fields))
 
-        query_lines = [f"SELECT {select_columns}", f"FROM {self.table_name}"]
-        # TODO expand functionality when needed (where, order_by, limit, etc)
+        query_lines = []
+        query_lines.append(f"SELECT {select_columns}")
+        query_lines.append(f"FROM {self.table_name}")
+        query_lines.append(self.where)
+        query_lines.append(self.order_by)
+        query_lines.append(self.limit)
+
+        return "\n".join([line for line in query_lines if line != ""])
 
-        return "\n".join(query_lines)
+
+class SalesforceDocBuilder:
+    def __init__(self, base_url):
+        self.base_url = base_url
+
+    def build_account(self, account):
+        owner = account.get("Owner", {})
+
+        opportunities = account.get("Opportunities")
+        opportunity_records = opportunities.get("records", []) if opportunities else []
+        opportunity = opportunity_records[0] if len(opportunity_records) > 0 else {}
+        opportunity_url = (
+            f"{self.base_url}/{opportunity.get('Id')}" if opportunity else ""
+        )
+        opportunity_status = opportunity.get("StageName", "")
+
+        return {
+            "_id": f"salesforce_connector_{account.get('Id')}",
```

Comment:
Is the prefix "salesforce_connector_" necessary here?

Also, are ids between accounts, leads, etc unique?

Reply:
> Is the prefix "salesforce_connector_" necessary here?

The ids for the ws connector documents are all `salesforce_<sobject_id>`. 

> 

They are apparently. You can use the id without any previous url endpoint to access a unique object. [source](https://developer.salesforce.com/forums/?id=9060G000000UVpzQAG)

Reply:
Oh that's good to know, TIL!

About ids - I think we don't need prefix here, and can just store `sobject_id`